src/java.base/share/classes/java/time/Clock.java

```diff
@@ -99,45 +101,18 @@
  *    }
  *  }
  * </pre>
- * This approach allows an alternate clock, such as {@link #fixed(Instant, ZoneId) fixed}
+ * This approach allows an alternative clock, such as {@link #fixed(Instant, ZoneId) fixed}
  * or {@link #offset(Clock, Duration) offset} to be used during testing.
  * <p>
  * The {@code system} factory methods provide clocks based on the best available
- * system clock This may use {@link System#currentTimeMillis()}, or a higher
+ * system clock. This may use {@link System#currentTimeMillis()}, or a higher
  * resolution clock if one is available.
  *
- * @implSpec
- * This abstract class must be implemented with care to ensure other classes operate correctly.
- * All implementations that can be instantiated must be final, immutable and thread-safe.
- * <p>
- * The principal methods are defined to allow the throwing of an exception.
- * In normal use, no exceptions will be thrown, however one possible implementation would be to
- * obtain the time from a central time server across the network. Obviously, in this case the
- * lookup could fail, and so the method is permitted to throw an exception.
- * <p>
- * The returned instants from {@code Clock} work on a time-scale that ignores leap seconds,
- * as described in {@link Instant}. If the implementation wraps a source that provides leap
- * second information, then a mechanism should be used to "smooth" the leap second.
- * The Java Time-Scale mandates the use of UTC-SLS, however clock implementations may choose
- * how accurate they are with the time-scale so long as they document how they work.
- * Implementations are therefore not required to actually perform the UTC-SLS slew or to
- * otherwise be aware of leap seconds.
- * <p>
- * Implementations should implement {@code Serializable} wherever possible and must
- * document whether or not they do support serialization.
- *
```

Comment:
The ImplSpec needs to say how it is implemented.
The 'implements InstantSource' can not mandate any particular implementation. Its just an interface the real behavior comes from its implementations.  In this case Clock.  Referring to the static methods of InstantSource behavior may be sufficient because that behavior is concrete.

Reply:
There are plenty of examples of interfaces in `java.time` and elsewhere that apply restrictions to implementations. Nevertheless, for simplicity and expediency I have reinstated the `implSpec` on `Clock`

Reply:
ok, thanks.
Assertions in interfaces are toothless. Tests can only be written for implementations.